dropwizard-auth/src/main/java/io/dropwizard/auth/AuthValueFactoryProvider.java

```diff
@@ -19,28 +16,31 @@
  * Value factory provider supporting {@link Principal} injection
  * by the {@link Auth} annotation.
  *
+ * Offers authentication of a single injected principal class. For
+ * support of polymorphic principals, see
+ * {@link PolymorphicAuthValueFactoryProvider}.
+ *
  * @param <T> the type of the principal
  */
 @Singleton
-public class AuthValueFactoryProvider<T extends Principal> extends AbstractValueFactoryProvider {
-
-    /**
-     * Class of the provided {@link Principal}
-     */
-    private final Class<T> principalClass;
+public class AuthValueFactoryProvider<T extends Principal>
+    extends PolymorphicAuthValueFactoryProvider
```

Comment:
I would prefer `AuthValueFactoryProvider` not to be extended from `PolymorphicAuthValueFactoryProvider`.

This ties these classes rather highly without a need. I think we can find another ways to reuse the common code, if this is a concern.